### ISSUE TITLE
[J4.x] Calendar field in subform cut off when overflowing subform container

### ISSUE THREAD

**Issue by crystalenka** [joomla/joomla-cms#38443]
### Steps to reproduce the issue
Create a calendar custom field.
Create a repeatable subform custom field.
Add the calendar field to the subform field. Save and close.

Open an article where you can edit the subform field and add some items.

### Expected result
You will be able to use the calendar field normally.

### Actual result
The calendar field gets cut off by the bounding box of the field container:

https://user-images.githubusercontent.com/9141288/184362896-e2af4357-b488-455e-9d08-0a8d9eac3d49.mov

### System information (as much as possible)
Joomla 4.1.5
PHP 8.0

### Additional comments
Possibly just a CSS bug, but I don't have the time to check it right now so I'm opening the issue. I might be able to come back later with a PR to fix it.

**Comment by crystalenka** [comment]
Some additional info:
- this happens on Safari, haven't tested in other browsers yet
- it appears to happen because of the `overflow-x:auto` rule on the `<div class="table-responsive">`. Removing that rule fixes the display of the calendar, but then the table overflows the container. 
- The best solution is probably to address the table behavior when it's too wide as horizontal scrolling is not a great UX, but that would also be a more involved solution.

**Comment by crystalenka** [comment]
This part of how bootstrap functions, apparently:

> ### Vertical clipping/truncation
> Responsive tables make use of overflow-y: hidden, which clips off any content that goes beyond the bottom or top edges of the table. In particular, this can clip off dropdown menus and other third-party widgets.

Source: https://getbootstrap.com/docs/5.2/content/tables/#responsive-tables 

So, unless we decide not to use bootstrap responsive tables, the solution is... impossible? I don't know. Maybe we could put the calendar select in a [dialog element](https://caniuse.com/dialog) (the relatively new native HTML modal)? 

However I think this will also be a problem for dropdown fields in subforms, possibly others, so it might be worth looking at in a broader context...

**Comment by chmst** [comment]
Related https://github.com/joomla/joomla-cms/issues/38346

**Comment by richard67** [comment]
Closing as having a pull request. Please test #39694 . Thanks in advance.

**Comment by crystalenka** [comment]
@richard67 probably should reopen. The PR does not address the issue fully as it's more complex than just a line of CSS; see my earlier comments in this issue.

**Comment by richard67** [comment]
@crystalenka You don’t have a button to reopen your issue when you visit it on GitHub directly and not in the issue tracker?

**Comment by crystalenka** [comment]
@richard67 I comment on mobile a lot so don't see it now (and didn't notice from the computer earlier, but could have missed it). Will take a look next time I'm on my computer. Thanks!

**Comment by 2003tanmay** [comment]
> @richard67 probably should reopen. The PR does not address the issue fully as it's more complex than just a line of CSS; see my earlier comments in this issue.

@crystalenka  yes you are right, it is more complicated than i thought.  I think I only have to fix the calendar, but as soon as I fixed the calendar it created another issue of the table overflowing the container.

**Comment by 2003tanmay** [comment]
@crystalenka @richard67 The Field has two "predefined" layouts for displaying the subform as either a table or as a div container, as well as support for custom layouts.
![Screenshot (64)](https://user-images.githubusercontent.com/91016907/218244418-33d15f45-40c3-4a9d-b2cc-920b9410d442.png)

When we have more than 4 fields then subform display as a div container.
![Screenshot (60)](https://user-images.githubusercontent.com/91016907/218244166-fad2dc02-00ef-442f-ae1c-bacbcc5e44c8.png)

And when we have 4 or less than 4 fields then subform display as a table.
![Screenshot (62)](https://user-images.githubusercontent.com/91016907/218244213-0f9efc5c-423c-40c2-9ff7-db1f3bf3fd8b.png)

"div container" does not cause issue of calendar and it not overflow horizontally. "table" causing the issue of calendar and when we fix the calendar by applying `overflow: visible`  it causes the table to overflow horizontally. So can we remove table layout and make div container layout as a default layout. I am unable to locate a file which has the logic behind when we have 4 or less then 4 field it display as a table otherwise it display as a div container. If this can be a possible solution to this issue can you please help me to find out the file location for this solution.

**Comment by 2003tanmay** [comment]
> @crystalenka @richard67 The Field has two "predefined" layouts for displaying the subform as either a table or as a div container, as well as support for custom layouts. ![Screenshot (64)](https://user-images.githubusercontent.com/91016907/218244418-33d15f45-40c3-4a9d-b2cc-920b9410d442.png)
> 
> When we have more than 4 fields then subform display as a div container. ![Screenshot (60)](https://user-images.githubusercontent.com/91016907/218244166-fad2dc02-00ef-442f-ae1c-bacbcc5e44c8.png)
> 
> And when we have 4 or less than 4 fields then subform display as a table. ![Screenshot (62)](https://user-images.githubusercontent.com/91016907/218244213-0f9efc5c-423c-40c2-9ff7-db1f3bf3fd8b.png)
> 
> "div container" does not cause issue of calendar and it not overflow horizontally. "table" causing the issue of calendar and when we fix the calendar by applying `overflow: visible` it causes the table to overflow horizontally. So can we remove table layout and make div container layout as a default layout. I am unable to locate a file which has the logic behind when we have 4 or less then 4 field it display as a table otherwise it display as a div container. If this can be a possible solution to this issue can you please help me to find out the file location for this solution.

@crystalenka can you please answer, this can be the possible solution for this issue or not?

**Comment by Fedik** [comment]
>  So can we remove table layout and make div container layout as a default layout.

We should not remove it. 
As you found, the fix for the issue is to use default `repeatable` layout instead of table layout, when there  `calendar` or `fancy-select` fields in subform.

And this will be possible in 4.3, #38180 added posibility to change layout.
This issue can be cloed I guess.
